packages/toolpad-app/src/toolpadDataSources/rest/client.tsx

```diff
@@ -200,6 +202,59 @@ function ConnectionParamsInput({ value, onChange }: ConnectionEditorProps<RestCo
   );
 }
 
+const isCorrectlyTransformedData = (preview: FetchResult) => {
+  const { data, untransformedData } = preview;
+
+  if (isEmpty(untransformedData)) {
+    return true;
+  }
+
+  return !isEmpty(data);
```

Comment:
Sorry it's not really clear what you are proposing or what's exactly the issue?

> Similarly, untransformedData can perfectly return an empty array while the transformed data returns undefined.

Can you provide an example?

Reply:
> Can you provide an example?

Sure thing, here's a transform function that legitimately returns an empty array:

use url: https://httpbin.org/json
use transform: `return data.slideshow.slides.filter(slide => slide.title.includes('foo'));`

I believe this should not show the warning.

I also think we should accept `null` as a legitimate value and only show a warning when the transform is returning `undefined`.
We should probably also make the `JsonView` show `undefined` instead of nothing when the input is `undefined`

_Lastly, I believe we should also consider preventing the overuse of basic `lodash` methods through `no-restricted-imports` and only whitelist a few like `pick`, `throttle`, `get` and `set`. Methods like `isEmpty` cover a too broad range of values and result in bugs like these. Also, the word "empty" doesn't have a well defined meaning in programming, making the code less intuitive. Better to use native constructs to express the intent clearer to other maintainers_

Reply:
Thanks, that's great feedback, I've addressed it in a follow up PR: https://github.com/mui/mui-toolpad/pull/1165